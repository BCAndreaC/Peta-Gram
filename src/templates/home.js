export const home = (onNavigate) => {
  // const container = document.querySelector('#container-content');
  const sectionhome = document.createElement('section');
  const articleHomeImage = document.createElement('article');
  const imgHome = document.createElement('img');
  const articleLogo = document.createElement('article');
  const imgLogo = document.createElement('img');
  const articleHomeDescription = document.createElement('article');
  const h1HomeDescription = document.createElement('h1');
  const spanPeta = document.createElement('span');
  const spanGram = document.createElement('span');
  const h2HomeDescription = document.createElement('h2');
  const articleBtnContainer = document.createElement('article');
  const btnSignUpHome = document.createElement('button');
  const btnSignInHome = document.createElement('button');

  sectionhome.classList.add('home-section');
  articleHomeImage.classList.add('home-image');
  articleLogo.classList.add('logo');
  articleHomeDescription.classList.add('home-description');
  articleBtnContainer.classList.add('btn-container');
  btnSignUpHome.classList.add('btnSignUp');
  btnSignInHome.classList.add('btnSignIn');
  spanPeta.classList.add('peta');
  spanGram.classList.add('gram');

  imgHome.setAttribute('alt', 'home-img');
  imgHome.setAttribute('src', 'images/woman-cat-dog.png');
  btnSignUpHome.setAttribute('type', 'button');
  btnSignInHome.setAttribute('type', 'button');
  imgLogo.setAttribute('alt', 'logo-img');
  imgLogo.setAttribute('src', 'images/logo.png');

  btnSignUpHome.textContent = 'Registrate';
  btnSignInHome.textContent = 'Ingresa';
  spanPeta.textContent = 'Peta';
  spanGram.textContent = 'gram';
  h1HomeDescription.textContent = 'Bienvenidx a ';
  h2HomeDescription.textContent = 'Nuestra comunidad está dedicada a crear un espacio seguro y acogedor para que todos puedan compartir sus experiencias y celebrar el amor y la alegria que nuestras mascotas traen a nuestras vidas. ¡Únete a nosotros y comparte tu amor hoy mismo!';
  btnSignInHome.addEventListener('click', () => {
    onNavigate('/signin');
  });
  btnSignInHome.addEventListener('click', () => {
    onNavigate('/signup');
  });

  articleHomeImage.appendChild(imgHome);
  articleLogo.appendChild(imgLogo);
  h1HomeDescription.appendChild(spanPeta);
  h1HomeDescription.appendChild(spanGram);
  articleHomeDescription.appendChild(h1HomeDescription);
  articleHomeDescription.appendChild(h2HomeDescription);
  articleBtnContainer.appendChild(btnSignInHome);
  articleBtnContainer.appendChild(btnSignUpHome);
  articleHomeDescription.appendChild(articleBtnContainer);
  sectionhome.appendChild(articleHomeImage);
  sectionhome.appendChild(articleLogo);
  sectionhome.appendChild(articleHomeDescription);

  return sectionhome;
};

// section.innerHTML = `
//   <article class="home-image">
//     <img src="./images/woman-cat-dog.png">
//   </article>
//   <article class="logo">
//     <img src="images/logo.png" alt="logo"/>
//   </article>
//   <article class="home-description">
//     <h1>Bienvenidx a <span class="peta">Peta</span><span class="gram">gram</span></h1>
// eslint-disable-next-line max-len
//     <h2>Nuestra comunidad está dedicada a crear un espacio seguro y acogedor para que todos puedan compartir sus experiencias y celebrar el amor y la alegria que nuestras mascotas traen a nuestras vidas. ¡Únete a nosotros y comparte tu amor hoy mismo!</h2>
//     <article class="btn-container">
//       <button type="button" class="btnSignUp">Registrate</button>
//       <button type="button" class="btnSignIn">Ingresa</button>
//     </article>
//   </article>
// `;

// const btnSignIn = section.querySelector('.btnSignIn');
// const btnSignUp = section.querySelector('.btnSignUp');
// const article = section.querySelector('.btn-container');
// const articleHome = section.querySelector('.home-description');
// article.appendChild(btnSignUp);
// article.appendChild(btnSignIn);

// articleHome.appendChild(article);
// container.appendChild(section);
