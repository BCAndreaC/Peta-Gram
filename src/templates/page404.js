export const page404 = () => {
  const section404 = document.createElement('section');
  const articleCat404 = document.createElement('article');
  const imgCat404 = document.createElement('img');
  const articleDescription404 = document.createElement('article');
  const h1Description404 = document.createElement('h1');
  const articleBtnContainer404 = document.createElement('article');
  const btn404 = document.createElement('button');
  const articleDog404 = document.createElement('article');
  const imgDog404 = document.createElement('img');

  section404.classList.add('pageNotFound');
  articleCat404.classList.add('cat404');
  articleDescription404.classList.add('description-pageNotFound');
  articleBtnContainer404.classList.add('container-btn_pageNotFound');
  btn404.classList.add('btn-pageNotfound');
  articleDog404.classList.add('dog404');

  imgCat404.setAttribute('src', './images/cat_bookshelf.png');
  imgCat404.setAttribute('alt', 'cat404');
  h1Description404.textContent = 'Woops! 404 Page not found';
  btn404.setAttribute('type', 'button');
  btn404.textContent = 'Home';
  imgDog404.setAttribute('src', 'images/dog_404.png');
  imgDog404.setAttribute('alt', 'dog404');

  articleCat404.appendChild(imgCat404);
  articleDescription404.appendChild(h1Description404);
  articleBtnContainer404.appendChild(btn404);
  articleDog404.appendChild(imgDog404);

  section404.appendChild(articleCat404);
  section404.appendChild(articleDescription404);
  section404.appendChild(articleBtnContainer404);
  section404.appendChild(articleDog404);
  // const container = document.querySelec;tor('#container-content');

  // const section = document.createElement('section');
  // section.classList.add('home-section');
  // section.innerHTML = `
  // <section class="pageNotFound">
  //   <article class="cat404">
  //     <img src="./images/cat_bookshelf.png" alt="cat404"/>
  //   </article>
  //       <article class ="description-pageNotFound">
  //           <h1>Woops! 404 Page not found</h1>
  //       </article>
  //       <article class="container-btn_pageNotFound">
  //       <button type="button" class="btn-pageNotfound">Home</button>
  //       </article>
  //       <article class="dog404">
  //           <img src="images/dog_404.png" alt="dog404"/>
  //       </article>
  //   </section>
  // `;
  return section404;
};
