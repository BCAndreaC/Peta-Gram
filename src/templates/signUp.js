export const signup = () => {
  const container = document.querySelector('#container-content');

  const section = document.createElement('section');
  section.classList.add('home-section');
  section.innerHTML = `
<section class="signUp-section" id="signUp-section">
 
  <form class="signUp-form">
    <a href="/" class="close">&times;</a>  <!-- &times es una entidad de caracteres HTML que representa el símbolo "×" -->
    <label class="form"> Nombre de Usuario </label>
    <input type="text" >
    <label class="form"> Correo Electronico </label> 
    <input type="email" id="userEmail">
    <label class="form"> Contraseña </label>
    <input type="password" id="userPassword">
    <button class="btnRegister" type="submit">Registrarse</button>
    <article class="return-home">
      <p>¿Ya tienes cuenta?</p> <a href="#">Ingresa aqui</a>
    </article>
  </form>
    
</section>`;

  const btnRegister = section.querySelector('.btnRegister');
  const div = document.createElement('div');
  const form = section.querySelector('.signUp-form');
  const close = section.querySelector('.close');

  form.appendChild(btnRegister);
  form.appendChild(close);
  div.appendChild(section);

  container.appendChild(div);

  return container;
};
