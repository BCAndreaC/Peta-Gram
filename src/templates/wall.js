export const wall = () => {
  const wallSection = document.createElement('section');
  const wallHeader = document.createElement('header');
  const hamburgerContainer = document.createElement('article');
  const hamburgerImg = document.createElement('img');
  const logoContainer = document.createElement('article');
  const logoImgContainer = document.createElement('article');
  const logoImg = document.createElement('img');
  const logoSpanContainer = document.createElement('article');
  const peta = document.createElement('span');
  const gram = document.createElement('span');
  const profileContainer = document.createElement('article');
  const profileImg = document.createElement('img');
  const wallMain = document.createElement('main');
  const postInputSection = document.createElement('section');
  const postArticle = document.createElement('article');
  const postForm = document.createElement('form');
  const inputLabel = document.createElement('label');
  const postInput = document.createElement('input');
  const postBtn = document.createElement('button');
  const postsSection = document.createElement('section');
  const postsArticle = document.createElement('article');
  const postHeaderDiv = document.createElement('div');
  const userImage = document.createElement('img');
  const userNameHeader = document.createElement('p');
  const postContent = document.createElement('div');
  const postText = document.createElement('p');
  const postFooter = document.createElement('div');
  const reactions = document.createElement('div');
  const dogReaction = document.createElement('img');
  const dogReactionCount = document.createElement('p');
  const catReaction = document.createElement('img');
  const catReactionCount = document.createElement('p');
  const commentPostBtn = document.createElement('button');

  const userImage2 = document.createElement('img');
  /* const userNameHeader2 = document.createElement('p'); */
  const postContent2 = document.createElement('div');
  /* const postText2 = document.createElement('p'); */

  wallSection.classList.add('wall-section');
  hamburgerContainer.classList.add('hamburger');
  logoContainer.classList.add('logo-container');
  logoImgContainer.classList.add('wall-logo');
  logoSpanContainer.classList.add('logo-span');
  peta.classList.add('peta');
  gram.classList.add('gram');
  profileContainer.classList.add('profilePic');
  wallMain.classList.add('container-wall');
  postInputSection.classList.add('post-input');
  postArticle.classList.add('your-post');
  postInput.classList.add('your-postinput');
  postBtn.classList.add('btnyour-post');
  postsSection.classList.add('posts-section');
  postsArticle.classList.add('post');
  postHeaderDiv.classList.add('post-header');
  userImage.classList.add('user-image');
  postContent.classList.add('post-content');
  postFooter.classList.add('post-bottom');
  reactions.classList.add('reactions');
  commentPostBtn.classList.add('btnComment');
  userImage2.classList.add('user-image');
  postContent2.classList.add('post-content');

  hamburgerImg.setAttribute('src', 'images/hamburguer.png');
  hamburgerImg.setAttribute('alt', 'hamburger');
  logoImg.setAttribute('src', 'images/logo.png');
  logoImg.setAttribute('alt', 'logo');
  logoImg.setAttribute('width', '83px');
  logoImg.setAttribute('height', '77px');
  profileImg.setAttribute('src', 'images/profile-img.png');
  profileImg.setAttribute('alt', 'profile picture');
  inputLabel.setAttribute('for', 'your-postinput');

  peta.textContent = 'Peta';
  gram.textContent = 'gram';

  hamburgerContainer.appendChild(hamburgerImg);
  logoImgContainer.appendChild(logoImg);
  logoSpanContainer.appendChild(peta);
  logoSpanContainer.appendChild(gram);
  logoContainer.appendChild(logoImgContainer);
  logoContainer.appendChild(logoSpanContainer);
  profileContainer.appendChild(profileImg);
  wallHeader.append(hamburgerContainer, logoContainer, profileContainer);
  //  main-section post input

  postForm.append(inputLabel, postInput, postBtn);
  postArticle.appendChild(postForm);
  postInputSection.appendChild(postArticle);

  //  post section

  postHeaderDiv.append(userImage, userNameHeader);
  postContent.appendChild(postText);

  reactions.append(dogReaction, dogReactionCount, catReaction, catReactionCount);
  postFooter.appendChild(reactions);
  postFooter.appendChild(commentPostBtn);
  postsArticle.appendChild(postHeaderDiv, postContent, postFooter);
  wallMain.append(postInputSection, postInputSection);
  wallSection.append(wallHeader, wallMain);
  return wallSection;
};
