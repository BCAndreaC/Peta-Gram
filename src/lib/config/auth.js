/* eslint-disable no-unused-vars */
import {
  signInWithPopup,
  GoogleAuthProvider,
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  sendEmailVerification,
  updateProfile,
  signOut,
} from 'firebase/auth';
import { auth } from './firebaseconfig.js';

export const login = (onNavigate, userEmail, userPassword, emailError, passwordError) => {
  const email = userEmail.value;
  const password = userPassword.value;
  emailError.textContent = '';
  passwordError.textContent = '';
  signInWithEmailAndPassword(auth, email, password)
    .then((userCredential) => {
    // Signed in
      const user = userCredential.user;
      onNavigate('/wall');
    // ...
    })
    .catch((error) => {
      const errorCode = error.code;
    });
};

export function registerUserWithEmailAndPassword(email, password, username) {
  return createUserWithEmailAndPassword(auth, email, password)
    .then((userCredential) => {
      const user = userCredential.user;
      sendEmailVerification(user);
      return updateProfile(auth.currentUser, {
        displayName: username,
      })
        .catch((error) => {
          const errorCode = error.code;
        });
    });
}

export function signOutUser() {
  return signOut(auth)
    .then(() => {
    }).catch((error) => {
      const errorSignOut = error.message;
    });
}

const provider = new GoogleAuthProvider();
export const googleLogin = (onNavigate) => {
  signInWithPopup(auth, provider)
    .then((result) => {
    // This gives you a Google Access Token. You can use it to access the Google API.
      const credential = GoogleAuthProvider.credentialFromResult(result);
      const token = credential.accessToken;
      // The signed-in user info.
      const user = result.user;
      onNavigate('/wall');
    // IdP data available using getAdditionalUserInfo(result)
    // ...
    }).catch((error) => {
    // Handle Errors here.
      const errorCode = error.code;
      const errorMessage = error.message;
      // The email of the user's account used.
      const email = error.customData.email;
      // The AuthCredential type that was used.
      const credential = GoogleAuthProvider.credentialFromError(error);
    // ...
    });
};
