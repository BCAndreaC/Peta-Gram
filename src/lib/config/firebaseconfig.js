import { initializeApp } from "firebase/app";
import { getAuth, onAuthStateChanged } from "firebase/auth";

const firebaseConfig = initializeApp({
    apiKey: "AIzaSyD8_N9eOlWmhT4UCsQi8wWYF5YL4xemIkU",
    authDomain: "petagram-b6b9a.firebaseapp.com",
    projectId: "petagram-b6b9a",
    storageBucket: "petagram-b6b9a.appspot.com",
    messagingSenderId: "628725489575",
    appId: "1:628725489575:web:107915c2aca03118352082",
    measurementId: "G-VL5GN2LXKX"
  });

const auth = getAuth(firebaseConfig);
onAuthStateChanged(auth, user => {
    if (user !== null){
        console.log("logged in!")
    } else {
        console.log("no user");
    }
})