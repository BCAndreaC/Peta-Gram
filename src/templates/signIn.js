import { googleLogin } from '../lib/config/auth';

export const signIn = (onNavigate) => {
  const signInSection = document.createElement('section');
  const articleForm = document.createElement('article');
  const signInForm = document.createElement('form');
  const emailLabel = document.createElement('label');
  const emailInput = document.createElement('input');
  const spanErrorEmail = document.createElement('span');
  const passwordLabel = document.createElement('label');
  const passwordInput = document.createElement('input');
  const spanErrorPass = document.createElement('span');
  const btnContainerSignUp = document.createElement('article');
  const btnLogin = document.createElement('button');
  const btnGoogle = document.createElement('button');
  const containerLinkSignIn = document.createElement('article');
  const askingText = document.createElement('p');
  const linkToSignUp = document.createElement('a');

  articleForm.classList.add('container-form');
  signInSection.classList.add('signIn-section');
  signInForm.classList.add('signIn-form');
  emailLabel.classList.add('form');
  spanErrorEmail.classList.add('errorMessage');
  btnContainerSignUp.classList.add('btn-container_signIn');
  btnLogin.classList.add('btnLogin');
  btnGoogle.classList.add('btnGoogle');
  passwordLabel.classList.add('form');
  containerLinkSignIn.classList.add('btn-container_signIn');

  emailInput.setAttribute('type', 'email');
  emailInput.setAttribute('id', 'loginUserEmail');
  passwordInput.setAttribute('type', 'password');
  passwordInput.setAttribute('id', 'loginUserPassword');
  spanErrorEmail.setAttribute('id', 'errorEmailMessage');
  spanErrorPass.setAttribute('id', 'errorPassMessage');
  linkToSignUp.setAttribute('id', 'link-signIn');
  btnLogin.setAttribute('type', 'submit');
  btnGoogle.setAttribute('type', 'button');

  emailLabel.textContent = 'Correo Electronico';
  passwordLabel.textContent = 'Contraseña';
  btnLogin.textContent = 'Ingresa';
  btnGoogle.textContent = 'Ingresa con Google';
  askingText.textContent = '¿No tienes cuenta? ';
  linkToSignUp.textContent = 'Registrate aquí';

  btnGoogle.addEventListener('click', () => {
    googleLogin(onNavigate);
  });

  linkToSignUp.addEventListener('click', () => {
    onNavigate('/signup');
  });

  signInForm.appendChild(emailLabel);
  signInForm.appendChild(emailInput);
  signInForm.appendChild(passwordLabel);
  signInForm.appendChild(passwordInput);
  signInForm.appendChild(spanErrorEmail);
  signInForm.appendChild(spanErrorPass);
  btnContainerSignUp.appendChild(btnGoogle);
  btnContainerSignUp.appendChild(btnLogin);
  containerLinkSignIn.appendChild(askingText);
  containerLinkSignIn.appendChild(linkToSignUp);

  signInForm.appendChild(articleForm);
  signInForm.appendChild(btnContainerSignUp);
  signInForm.appendChild(containerLinkSignIn);

  signInSection.appendChild(signInForm);
  return signInSection;
};

//   section.innerHTML = `

//   <form class="signIn-form">

//     <label class="form"> Correo Electronico </label>
//     <input type="email" id="loginUserEmail">
//     <span id="errorEmailMessage" class="errormessage"></span>
//     <label class="form"> Contraseña </label>
//     <input type="password" id="loginUserPassword">
//     <span id="errorPassMessage" class="errormessage"></span>
//     <article class="btn-container_signIn">
//       <button class="btnLogin" type="submit">Ingresa</button>
//       <button class="btnGoogle" type="button"> Ingresa con Google </button>
//     </article>
//     <article class="return-home">
//       <p>¿No tienes cuenta?</p><a href="#" id="link-signIn">Registrate aqui</a>
//     </article>
//   </form>

// </section>`;
