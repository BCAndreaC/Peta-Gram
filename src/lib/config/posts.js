import { addDoc, getDocs, serverTimestamp } from 'firebase/firestore';
import { colRef } from './firebaseconfig';

getDocs(colRef)
  .then((snapshot) => {
    const posts = [];
    snapshot.docs.forEach((doc) => {
      posts.push({ ...doc.data(), id: doc.id });
    });
    console.log(posts);
  })
  .catch((err) => {
    console.log(err.message);
  });

export function posting(input, form) {
  addDoc(colRef, {
    text: input.value,
    timestamp: serverTimestamp(),
  })
    .then(() => {
      form.reset();
    });
  console.log(getDocs(colRef));
}
