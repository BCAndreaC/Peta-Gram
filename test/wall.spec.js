/**
 * @jest-environment jsdom
 */
import { wall } from '../src/templates/wall';
import { signOutUser } from '../src/lib/config/auth';
import { posting, postData } from '../src/lib/config/posts';

jest.mock('firebase/auth', () => ({
  getAuth: () => ({
    currentUser: {
      displayName: 'John Doe',
    },
  }),
}));

jest.mock('../src/lib/config/auth', () => ({
  signOutUser: jest.fn(() => Promise.resolve()),

}));
jest.mock('../src/lib/config/posts', () => ({
  posting: jest.fn(),
  postData: jest.fn(),

}));

describe('wall', () => {
  let wallSection;

  beforeEach(() => {
    wallSection = wall();
  });

  afterEach(jest.clearAllMocks);

  it('deberia de crear los elementos nesarios para el muro', () => {
    expect(wallSection).toBeTruthy();
    expect(wallSection.querySelector('header')).toBeTruthy();
    expect(wallSection.querySelector('.hamburger')).toBeTruthy();
    expect(wallSection.querySelector('.logo-container')).toBeTruthy();
    expect(wallSection.querySelector('.profilePic')).toBeTruthy();
    expect(wallSection.querySelector('main')).toBeTruthy();
    expect(wallSection.querySelector('.post-input')).toBeTruthy();
    expect(wallSection.querySelector('.posts-section')).toBeTruthy();
  });

  it('debe navegar a Home al hacer click en el boton cerrar sesión', async () => {
    const onNavigate = jest.fn();
    const container = document.createElement('section');
    container.append(wall(onNavigate));

    const closeBtn = container.querySelector('.btn-logout');
    expect(closeBtn).not.toBe(null);
    // Click the close button
    closeBtn.click();
    await signOutUser.mockImplementation(() => Promise.resolve());
    expect(onNavigate).toHaveBeenCalledWith('/');
  });

  test('Si el input no está vacío debe llamarse la función posting', () => {
    const container = document.createElement('section');
    container.append(wall());
    const form = container.querySelector('.form-yourpost');
    const input = container.querySelector('.your-postInput');
    input.value = 'Some post text';

    form.dispatchEvent(new Event('submit', { bubbles: true }));

    expect(posting).toHaveBeenCalledWith(input, form);
  });

  test('al hacer clic en el menú hamburguesa la clase debe cambiar a active', () => {
    const container = document.createElement('section');
    container.append(wall());
    const hamburgerArticle = container.querySelector('.hamburger');
    const sectionMenu = document.createElement('section-menu');

    hamburgerArticle.dispatchEvent(new Event('click'));
    expect(sectionMenu.classList.contains('active')).toBe(false);

    expect(sectionMenu.classList.contains('active')).toBe(false);
  });

  it('should render posts correctly', () => {
    const mockQuerySnapshot = {
      forEach: jest.fn((callback) => {
        const mockPostData = {
          userid: 'user1',
          text: 'Some post text',
          likes: [],
          likescat: [],
        };
        const mockPos = {
          id: 'post1',
          data: () => mockPostData,
        };
        callback(mockPos);
      }),
    };

    postData.mockImplementationOnce((callback) => {
      callback(mockQuerySnapshot);
    });
    wall();
  });
});
