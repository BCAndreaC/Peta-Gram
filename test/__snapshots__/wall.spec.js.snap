// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wall deberia de renderizar la pantalla wall correctamente 1`] = `"<section class=\\"wall-section\\"><section class=\\"section-menu\\"><button class=\\"btn-close-menu\\">×</button><section class=\\"section-btn-menu\\"><section class=\\"section-perfil-menu\\"><button class=\\"btn-menu\\">Perfil<img class=\\"img-perfil-menu\\" src=\\"images/cat-porfile-menu.png\\" width=\\"60\\" height=\\"60\\"></button></section><section><button class=\\"btn-menu\\">Muro<img class=\\"img-muro-menu\\" src=\\"images/cat-muro-menu.png\\" width=\\"60\\" height=\\"60\\"></button></section><section><button class=\\"btn-menu\\">Amigxs<img class=\\"img-amigxs-menu\\" src=\\"images/amigxs-menu.png\\" width=\\"60\\" height=\\"60\\"></button></section></section><section class=\\"section-logout\\"><button class=\\"btn-logout\\">Cerrar sesión<img class=\\"img-signout\\" src=\\"images/signout-dog.png\\" width=\\"60\\" height=\\"60\\"></button></section></section><header><article class=\\"hamburger\\"><img src=\\"images/hamburguer.png\\" alt=\\"hamburguer\\"></article><article class=\\"logo-container\\"><article class=\\"wall-logo\\"><img src=\\"images/logo.png\\" alt=\\"logo\\" width=\\"83\\" height=\\"77\\"></article><article class=\\"logo-span\\"><span class=\\"peta\\"></span><span class=\\"gram\\"></span></article></article><article class=\\"profilePic\\"><img src=\\"images/profile-img.png\\" alt=\\"profile picture\\"></article></header><main class=\\"container-wall\\"><section class=\\"post-input\\"><article class=\\"your-post\\"><form class=\\"form-yourpost\\"><label for=\\"your-postinput\\"></label><br><input class=\\"your-postInput\\" type=\\"text\\" id=\\"your-postinput\\" name=\\"your-postinput\\" placeholder=\\"Escribe algo sobre tus mascotas...\\"><br><article class=\\"containerbutton\\"><button class=\\"btnyour-post\\"><img src=\\"images/pawn.png\\" alt=\\"pawn\\">Publicar</button></article></form></article></section><section class=\\"posts-section\\"><article class=\\"post\\"><div class=\\"post-header\\"><img class=\\"user-image\\" src=\\"images/user1.png\\" alt=\\"profile picture\\"><p>Fulanitx escribió:</p></div><div class=\\"post-content\\"><p>¡Hola a todxs! Dale like si a tu mascota le gusta el pollito.</p></div><div class=\\"post-bottom\\"><div class=\\"reactions\\"><img src=\\"images/dog.png\\" alt=\\"dog reaction\\"><p>10</p><img src=\\"images/cat.png\\" alt=\\"cat reaction\\"><p>18</p></div><button class=\\"btnComment\\" type=\\"button\\"><img src=\\"images/post-pawn.png\\">Comentar</button></div></article><article class=\\"post\\"><div class=\\"post-header\\"><img class=\\"user-image\\" src=\\"images/user2.png\\" alt=\\"profile picture\\"><p>Fulanitx escribió:</p></div><div class=\\"post-content\\"><p>Les recomiendo estos lugares pet-friendly: Patazonia, TacoFriends y Cinemalitos.</p></div><div class=\\"post-bottom\\"><div class=\\"reactions\\"><img src=\\"images/dog.png\\" alt=\\"dog reaction\\"><p>2</p><img src=\\"images/cat.png\\" alt=\\"cat reaction\\"><p>18</p></div><button class=\\"btnComment\\" type=\\"button\\"><img src=\\"images/post-pawn.png\\">Comentar</button></div></article></section></main><div class=\\"confirmationModal\\" style=\\"display: none;\\"><p>¿Segurx que deseas eliminar esta publicación?</p><img src=\\"images/dogConfirm.png\\"><button class=\\"confirmBtn\\">Eliminar</button><button class=\\"cancelBtn\\">Cancelar</button></div></section>"`;
